app/controllers/models_controller.rb

```diff
@@ -0,0 +1,15 @@
+class ModelsController < ApplicationController
+  include Pagy::Backend
+
+  def index
+    @pagy, @models = pagy(Model.ordered_by_vehicle_count)
+
+    @models = @models.map do |model|
+      model.define_singleton_method(:model_image) do
+        images = model.vehicle_details.map(&:image)
+        images.find(&:attached?)
+      end
+      model
```

Comment:
Sao lại cần phải viết như này nhỉ 
giải thích cho anh thuật toán em xử lý ở đây nhé